ui/components/app/clear-metametrics-data/clear-metametrics-data.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useContext } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  hideDeleteMetaMetricsDataModal,
+  markingMetaMetricsDataDeletion,
+  openDataDeletionErrorModal,
+} from '../../../ducks/app/app';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  Box,
+  Button,
+  ButtonSize,
+  ButtonVariant,
+  Modal,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from '../../component-library';
+import {
+  AlignItems,
+  BlockSize,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import {
+  createMetaMetricsDataDeletionTask,
+  setHasMetaMetricsDataRecorded,
+} from '../../../store/actions';
+import { getParticipateInMetaMetrics } from '../../../selectors';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+
+export default function ClearMetaMetricsData() {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const trackEvent = useContext(MetaMetricsContext);
+
+  const participateInMetaMetrics = useSelector(getParticipateInMetaMetrics);
+
+  const closeModal = () => {
+    dispatch(hideDeleteMetaMetricsDataModal());
+  };
+
+  const deleteMetaMetricsData = async () => {
+    try {
+      await createMetaMetricsDataDeletionTask();
+      dispatch(markingMetaMetricsDataDeletion());
+      if (participateInMetaMetrics) {
```

Comment:
As discussed on call, `trackEvent` does not need to go inside this if block, as `participateInMetaMetrics` is checked at the controller level